File: CPP/OOP/pointMain.cpp
#include <iostream>

#include "Point.h"

using namespace std;

int main() {
	Point* a = new Point(3, 4);
	Point* b = new Point;
	
	cout << "Point a: " << *a << endl;
	cout << "Point b: " << *b << endl;

	if ( *a != *b ) {
		cout << "Distance is " << a->getDistance(*b) << endl;
	} else {
		cout << "Perfectly equal!" << endl;
	}


	return 0;
}

File: OtherLang/PointMain.java
public class PointMain {
	public static void main(String[] args) {
		Point a = new Point(3, 4);
		Point b = new Point();

		System.out.println("distance: " + a.getDistance(b));
		System.out.print(a);

		if ( a.equals(b) ) {
			System.out.print(" equals ");
		} else {
			System.out.print(" not equal ");
		}

		a = Null;
		
		System.out.println(b);		
	}
}